test/prism_regression/lambda.rb

```diff
@@ -3,7 +3,9 @@
 -> { 123 }
 -> (param) { param }
 -> (optional_param = 123) { optional_param }
--> { 456 }.call
+# TODO: Legacy parser doesn't include `->` in the location of lambda's `Block` node,
+# So the the receiver of this call will have the wrong start location.
+# -> { 456 }.call
```

Comment:
We can easily fix this in the "new world" where we stop creating legacy parse trees. https://github.com/sorbet/sorbet/issues/9631